lib/ecmarkdown.js

```diff
@@ -1,11 +1,357 @@
 'use strict';
-var parser = require('./generated-parser.js');
+
 var escapeHtml = require('escape-html');
+var beautify = require('./beautify.js');
 
-exports.list = function (ecmarkdown) {
-    return parser.parse(ecmarkdown.trim() + '\n', { startRule: 'list' });
+exports.document = function (ecmarkdown) {
+  return beautify(emitDocument(parse(ecmarkdown)));
 };
 
 exports.fragment = function (ecmarkdown) {
-    return parser.parse(escapeHtml(ecmarkdown.trim()), { startRule: 'fragment' });
+  return beautify(emitFragment(parse(escapeHtml(ecmarkdown))));
 };
+
+
+var lexemes = {
+  star: /\*/,
+  underscore: /_/,
+  tick: /`[^`\n]*`/,
```

Comment:
I will add a general comment about the grammar of this implementation. But to answer your question, star, underscore, and tilde can nest, whereas tick, string, and pipe cannot have other things inside of it. I alluded to this above - we could disallow nesting in all cases, but it seemed like something we might want in the future...